src/uct/ib/ud/base/ud_ep.c

```diff
@@ -667,14 +671,16 @@ uct_ud_ep_process_ack(uct_ud_iface_t *iface, uct_ud_ep_t *ep,
                       uct_ud_psn_t ack_psn, int is_async)
 {
     /* Ignore duplicate ACK */
-    if (ucs_unlikely(UCT_UD_PSN_COMPARE(ack_psn, <=, ep->tx.acked_psn))) {
+    if (ucs_unlikely(UCT_UD_PSN_COMPARE(ack_psn, <=, ep->tx.acked_psn) ||
+                     (ep->flags & UCT_UD_EP_FLAG_FLUSH_CANCEL))) {
```

Comment:
added a comment

Reply:
not clear -- after flush(CANCEL), ep->tx.acked_psn = send.psn-1 - set by ud_ep_purge.
 since ack_psn <= send_psn-1 it means ack_psn <= ep->tx.acked_psn,

Reply:
ah, I see - removed